breathecode/events/tests/urls/tests_academy_eventype_slug_visibilitysetting.py

```diff
@@ -0,0 +1,201 @@
+            'name': visibility_setting['syllabus']['name'],
+            'slug': visibility_setting['syllabus']['slug'],
+        } if visibility_setting['syllabus'] else None,
+        **data,
+    }
+
+
+class AcademyEventTypeVisibilitySettingsTestSuite(EventTestCase):
+    cache = EventCache()
+
+    def test_post_event_type_with_no_auth(self):
+
+        url = reverse_lazy('events:academy_eventype_slug_visibilitysetting',
+                           kwargs={'event_type_slug': 'funny_event'})
+
+        response = self.client.get(url)
+        json = response.json()
+        expected = {'detail': 'Authentication credentials were not provided.', 'status_code': 401}
+
+        self.assertEqual(json, expected)
+        self.assertEqual(response.status_code, 401)
+
+    def test_get_visibilitysetting_with_bad_slug(self):
+        self.bc.request.set_headers(academy=1)
+
+        url = reverse_lazy('events:academy_eventype_slug_visibilitysetting',
+                           kwargs={'event_type_slug': 'funny_event'})
+        self.generate_models(
+            authenticate=True,
+            profile_academy=1,
+            role=1,
+            capability='read_event_type',
+        )
+
+        response = self.client.get(url)
+        json = response.json()
+        expected = {'detail': 'not-found', 'status_code': 400}
+
+        self.assertEqual(json, expected)
+        self.assertEqual(response.status_code, 400)
+
+    def test_get_visibilitysetting(self):
+        self.bc.request.set_headers(academy=1)
+
+        url = reverse_lazy('events:academy_eventype_slug_visibilitysetting',
+                           kwargs={'event_type_slug': 'funny_event'})
+        self.generate_models(
+            authenticate=True,
+            profile_academy=1,
+            role=1,
+            event_type_visibility_setting=True,
+            event_type={
+                'slug': 'funny_event',
+                'icon_url': 'https://www.google.com',
+                'visibility_settings': 1
+            },
+            capability='read_event_type',
+        )
+
+        response = self.client.get(url)
+        json = response.json()
+        expected = [get_serializer(vs) for vs in json]
+
+        self.assertEqual(json, expected)
+        self.assertEqual(response.status_code, 200)
+
+    def test_post_visibilitysetting_with_bad_slug(self):
+        self.bc.request.set_headers(academy=1)
+
+        url = reverse_lazy('events:academy_eventype_slug_visibilitysetting',
+                           kwargs={'event_type_slug': 'funny_event'})
+        self.generate_models(
+            authenticate=True,
+            profile_academy=1,
+            role=1,
+            capability='crud_event_type',
+        )
+
+        response = self.client.post(url)
+        json = response.json()
+        expected = {'detail': 'event-type-not-found', 'status_code': 400}
+
+        self.assertEqual(json, expected)
+        self.assertEqual(response.status_code, 400)
```

Comment:
Add test of database to the rest of tests

Reply:
So think you is modifying 2 tables, `EventTypeVisibilitySetting` and the handler between it and `EventType`